plugins/woocommerce-admin/client/products/sections/product-details-section.tsx

```diff
@@ -42,6 +43,10 @@ export const ProductDetailsSection: React.FC = () => {
 		}
 	);
 
+	function doesNameHaveError(): boolean {
+		return Boolean( touched.name ) && Boolean( errors.name );
```

Comment:
Future enhancement... form context should export functions like `isFieldTouched()` and `isFieldValid()`... feels nicer.

Reply:
> Why do you have to check for touched as well?

That's because when the form is mounted for the first time the values are invalid (because there no values or they are empty) but the input field does not show the error until the user touches it.

Reply:
Yeah, I felt this seemed less-than-ideal. Then I looked at my [WIP PR for scheduled sale fields](https://github.com/woocommerce/woocommerce/pull/34538) and realize that I'm doing the exact same sort of check.   😄